src/library/scala/math/BigDecimal.scala

```diff
@@ -416,11 +416,11 @@ extends ScalaNumber with ScalaNumericConversions with Serializable with Ordered[
       }
     case _                    => isValidLong && unifiedPrimitiveEquals(that)
   }
-  override def isValidByte  = noArithmeticException(toByteExact)
-  override def isValidShort = noArithmeticException(toShortExact)
+  override def isValidByte  = noArithmeticException{toByteExact; ()}
+  override def isValidShort = noArithmeticException{toShortExact; ()}
   override def isValidChar  = isValidInt && toIntExact >= Char.MinValue && toIntExact <= Char.MaxValue
-  override def isValidInt   = noArithmeticException(toIntExact)
-  def isValidLong  = noArithmeticException(toLongExact)
+  override def isValidInt   = noArithmeticException{toIntExact; ()}
+  def isValidLong  = noArithmeticException{toLongExact; ()}
```

Comment:
I like my fix https://github.com/scala/scala/pull/10093/files#diff-04cabf28eb65a0db612090426c220adc7e88f525bf3fd46cd210ef52e52b1f9aR462

Some code like Akka or Future uses `def f[U]` to imply visually that `U` is effectively `Unit` and will be discarded.